src/util.ts

```diff
@@ -15,3 +17,51 @@ export function arrayEquals<T>(arr1: Array<T>, arr2: Array<T>): boolean {
 export function isPermittedOrigin(url: URL): boolean {
   return url.protocol == 'https:' && arrayEquals(url.hostname.split('.').slice(-2), ['mongodb', 'com']);
 }
+
+function convertTitleCase(name: string): string {
+  return name.replace(/^[_-]*(.)|[_-]+(.)/g, (s, c, d) => (c ? c.toUpperCase() : ' ' + d.toUpperCase()));
+}
+
+/**
+ *
+ * @param taxonomy
+ * @returns a trie structure of taxonomy.
+ * each node is denoted by a 'name' attribute.
+ * other attributes denotes a new node
+ * ['name' and 'display_name' attributes denote name(s) of facet]
+ * [versions have special boolean attribute of 'stable']
+ */
+export function convertTaxonomyResponse(taxonomy: Taxonomy): FacetDisplayNames {
+  const res: FacetDisplayNames = {};
+
+  function addToRes(entityList: TaxonomyEntity[], ref: { [key: string]: any }, property: string) {
+    const conversion = property === 'versions' ? (s: string) => s : convertTitleCase;
+    ref[property] = {
+      name: convertTitleCase(property), // convert snakecase to title case
+    };
+    ref = ref[property];
+    for (const taxEntity of entityList) {
+      const entity: Record<string, any> = {
+        name: taxEntity['display_name'] || conversion(taxEntity['name']),
+      };
+      if (property === 'versions' && taxEntity['stable']) {
+        entity['stable'] = true;
+      }
+      for (const key in taxEntity) {
+        if (!Array.isArray(taxEntity[key])) {
+          continue;
+        }
+        addToRes(taxEntity[key] as TaxonomyEntity[], entity, key);
```

Comment:
I don't think we need the continue here - should be able to do something like this

```suggestion
        if (Array.isArray(taxEntity[key])) {
          addToRes(taxEntity[key] as TaxonomyEntity[], entity, key);
        }
```

Reply:
this was just preference of `if true, return early` (in this case `continue` since its a loop) (and a minor point of saving an indentation 😛 )

```
const fn = () => {
   if (true) {return;}
   doSomething();
   doAnother();
}
```
vs
```
const fn = () => {
   if (false) {
      doSomething()
      doAnother();
   }
}
```